src/Model/GroundWaterFlow/gwf3vsc8.f90

```diff
@@ -0,0 +1,1519 @@
+!
+!    SPECIFICATIONS:
+! ------------------------------------------------------------------------------
+    ! -- modules
+    ! -- dummy
+    class(GwfVscType), intent(inout) :: this
+    ! -- local
+    character(len=LINELENGTH) :: errmsg, keyword
+    integer(I4B) :: ierr
+    logical :: isfound, endOfBlock
+    ! -- format
+! ------------------------------------------------------------------------------
+    !
+    ! -- get dimensions block
+    call this%parser%GetBlock('DIMENSIONS', isfound, ierr, &
+                              supportOpenClose=.true.)
+    !
+    ! -- parse dimensions block if detected
+    if (isfound) then
+      write (this%iout, '(/1x,a)') 'PROCESSING VSC DIMENSIONS'
+      do
+        call this%parser%GetNextLine(endOfBlock)
+        if (endOfBlock) exit
+        call this%parser%GetStringCaps(keyword)
+        select case (keyword)
+        case ('NVISCSPECIES')
+          this%nviscspecies = this%parser%GetInteger()
+          write (this%iout, '(4x,a,i0)') 'NVISCSPECIES = ', this%nviscspecies
+        case default
+          write (errmsg, '(4x,a,a)') &
+            'UNKNOWN VSC DIMENSION: ', trim(keyword)
+          call store_error(errmsg)
+          call this%parser%StoreErrorUnit()
+        end select
+      end do
+      write (this%iout, '(1x,a)') 'END OF VSC DIMENSIONS'
+    else
+      call store_error('REQUIRED VSC DIMENSIONS BLOCK NOT FOUND.')
+      call this%parser%StoreErrorUnit()
+    end if
+    !
+    ! -- check dimension
+    if (this%nviscspecies < 1) then
+      call store_error('NVISCSPECIES MUST BE GREATER THAN ZERO.')
+      call this%parser%StoreErrorUnit()
+    end if
+    !
+    ! -- return
+    return
+  end subroutine read_dimensions
+
+  !> @ brief Read data for package
+  !!
+  !!  Method to read data for the package.
+  !!
+  !<
+  subroutine read_packagedata(this)
+    ! -- modules
+    ! -- dummy
+    class(GwfVscType) :: this
+    ! -- local
+    character(len=LINELENGTH) :: errmsg
+    character(len=LINELENGTH) :: line
+    integer(I4B) :: ierr
+    integer(I4B) :: iviscspec
+    logical :: isfound, endOfBlock
+    logical :: blockrequired
+    integer(I4B), dimension(:), allocatable :: itemp
+    character(len=10) :: c10
+    character(len=16) :: c16
+    ! -- format
+    character(len=*), parameter :: fmterr = &
+      "('INVALID VALUE FOR IRHOSPEC (',i0,') DETECTED IN VSC PACKAGE. &
+      &IRHOSPEC MUST BE > 0 AND <= NVISCSPECIES, AND DUPLICATE VALUES &
+      &ARE NOT ALLOWED.')"
+! ------------------------------------------------------------------------------
+    !
+    ! -- initialize
+    allocate (itemp(this%nviscspecies))
+    itemp(:) = 0
+    !
+    ! -- get packagedata block
+    blockrequired = .true.
+    call this%parser%GetBlock('PACKAGEDATA', isfound, ierr, &
+                              blockRequired=blockRequired, &
+                              supportOpenClose=.true.)
+    !
+    ! -- parse packagedata block
+    if (isfound) then
```

Comment:
what happens if packagedata is not found?

Reply:
Added an else statement.  Because VSC has a lot of parallel with BUY, I had a look at BUY and submitted a PR (#1077) for the same issue there.  Addressed in 736b861